launcher/launcher.go

```diff
@@ -1213,3 +1213,29 @@ func deviceAllocateAlert(instance *runtimeInstanceInfo, device string, err error
 		},
 	}
 }
+
+func instanceInfoEqual(info1, info2 *aostypes.InstanceInfo) bool {
+	if info1.InstanceIdent != info2.InstanceIdent ||
+		info1.NetworkParameters.Subnet != info2.NetworkParameters.Subnet ||
+		info1.NetworkParameters.IP != info2.NetworkParameters.IP ||
+		info1.NetworkParameters.VlanID != info2.NetworkParameters.VlanID ||
+		info1.UID != info2.UID ||
+		info1.Priority != info2.Priority ||
+		info1.StoragePath != info2.StoragePath ||
+		info1.StatePath != info2.StatePath {
+		return false
+	}
+
+next:
+	for _, dns1 := range info1.NetworkParameters.DNSServers {
```

Comment:
Not proper comparison: what if info2 has more items than info1? Compare lens first.

Reply:
Done